lib/runtime/juttle-utils.js

```diff
@@ -68,6 +70,63 @@ function filterPoints(points, from, to) {
     });
 }
 
+// Assigns the time from timeField to 'time' and attempts to convert it
+// into JuttleMoment
+//
+// If not specified, timeField defaults to 'time'. If specified and point
+// is missing that field, emits a warning using the supplied context.
+function toNativeTime(points, options) {
+    var self = this;
+
+    options = options || {};
+    var epsilon = options.epsilon;
+    var timeField = options.timeField;
+    var timeRenameField = options.timeRenameField;
+
+    points = _.map(points, function(pt) {
+        if (timeRenameField && typeof timeRenameField === 'string') {
+            pt[timeRenameField] = pt.time;
+            delete pt.time;
+        }
+
+        if (timeField) {
+            if (!pt[timeField]) {
+                self.trigger('warning', new errors.runtimeError('RT-POINT-MISSING-TIME', {field: timeField}));
+            }
+            if (timeField !== 'time') {
+                if (pt.time && !timeRenameField) {
+                    // if timeRenameField is truthy boolean or number => will suppress warning without adding a new field
+                    self.trigger('warning', new errors.runtimeError('RT-OVERWRITING-TIME', {field: timeField}));
+                }
+                pt.time = pt[timeField];
+                delete pt[timeField];
+            }
+        }
+
+        if (pt.time !== undefined && !(pt.time instanceof JuttleMoment)) {
+            var date = makeDate(pt.time);
+            if (date.getTime() === date.getTime()) { // speedy check for invalid date, to make sure its getTime() isn't NaN
+                pt.time = new JuttleMoment({rawDate: date});
+            } else {
+                try {
+                    pt.time = new JuttleMoment(pt.time);
+                } catch(e) {
+                    if (self.trigger) {
+                        self.trigger('warning', new errors.runtimeError('RT-INVALID-TIME-ERROR', {time: pt.time}));
+                    }
+                    return null;
+                }
```

Comment:
@davidvgalbraith we could move this warning to `toNative` and use `toNative` function. Does that work?